common4j/src/main/com/microsoft/identity/common/java/telemetry/events/CertBasedAuthEvent.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.telemetry.events;
+
+import com.microsoft.identity.common.java.exception.BaseException;
+import com.microsoft.identity.common.java.telemetry.TelemetryEventStrings;
+
+import lombok.NonNull;
+
+/**
+ * Event specifically for emitting certificate-based authentication (CBA) info.
+ */
+public class CertBasedAuthEvent extends BaseEvent {
+
+    /**
+     * Creates a new instance of CertBasedAuthEvent.
+     * @param eventName Telemetry string designating the type of CBA being proceeded with (on-device or smartcard).
+     */
+    public CertBasedAuthEvent(final String eventName) {
+        super();
+        names(eventName);
+        types(TelemetryEventStrings.EventType.CERT_BASED_AUTH_EVENT);
+    }
+
+    /**
+     * Puts a Boolean that describes whether or not a PivProvider instance was already present in the Security static provider list
+     *  at the time of proceeding with smartcard CBA.
+     * @param isPresent true when a PivProvider instance was already present. false otherwise.
+     * @return The Event object.
+     */
+    public CertBasedAuthEvent putIsExistingPivProviderPresent(final boolean isPresent) {
```

Comment:
These 3 methods are specific to yubikey, so I would suggest 

1. extending this class into "YubiKeyCertBasedAuthEvent"
or 
2. create a helper method somewher (maybe inside the challenge handler, that constructs the event based on the parameters).

Reply:
Based on comments earlier above, I'm wondering if it would be ok to put the `putIsExistingPivProviderPresent` and `putPivProviderRemoved` in their own event name (something like CERT_BASED_AUTH_YUBIKEY_PROVIDER_STATUS_EVENT) and then rename this class to `CertBasedAuthResultEvent` (with corresponding names CERT_BASED_AUTH_{SMARTCARD/ON_DEVICE}_RESULT).